src/bin/cargo/commands/help.rs

```diff
@@ -40,23 +40,20 @@ pub fn exec(gctx: &mut GlobalContext, args: &ArgMatches) -> CliResult {
         return Ok(());
     }
 
-    match check_builtin(&subcommand) {
-        Some(s) => {
-            crate::execute_internal_subcommand(gctx, &[OsStr::new(s), OsStr::new("--help")])?;
-        }
-        None => {
-            crate::execute_external_subcommand(
-                gctx,
-                subcommand,
-                &[OsStr::new(subcommand), OsStr::new("--help")],
-            )?;
-        }
+    if super::builtin_exec(subcommand).is_none() {
```

Comment:
nit: I think in a previous iteration, you had used `!option.is_some()`.  One advantage of that is it makes the fact that this is negative logic upfront.

Thoughts?

Reply:
Also, is there a reason you swapped the order of the cases instead of doing `option.is_some()`?